src/System.Composition.Convention/src/Microsoft/Internal/Requires.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.Internal
+{
+    internal static class Requires
+    {
```

Comment:
@KrzysztofCwalina I think `Requires` is meant to specify preconditions (which someone calling your code may violate), and 'Assumes" is meant to specify invariants, which if they fail means there's a bug in the library code (in this case System.Composition).